src/ripple/app/paths/Pathfinder.cpp

```diff
@@ -181,33 +199,23 @@ Pathfinder::Pathfinder (
         mDstAmount (saDstAmount),
         mSrcCurrency (uSrcCurrency),
         mSrcIssuer (uSrcIssuer),
-        mSrcAmount ({uSrcCurrency, uSrcIssuer}, 1u, 0, true),
+        convert_all_ (mDstAmount ==
+            STAmount (mDstAmount.issue (), 1u, 0, true)),
```

Comment:
@rec I had am STAmount method that checked precisely what you and and Ed mentioned and then I came to the same conclusion Nik mentioned.